Python/compile.c

```diff
@@ -7744,6 +7748,12 @@ assemble(struct compiler *c, int addNone)
         }
     }
 
+    for (basicblock *b = c->u->u_blocks; b != NULL; b = b->b_list) {
```

Comment:
I am confused on why this works. Aren't the blocks supposed to be topologically linked already?

Reply:
We haven't computed the entry block yet, so we use `b_list`.

It also happens that using the backwards order gives better results.
Jumps to exits tend to be forward, so going backwards will be able to combine chains of jumps that going forwards would not.

I'm not making any claims that it is complete, it may well miss some blocks, but it should be good enough.